### ISSUE TITLE
Option to output a generated sync function as a single-line JSON string

### ISSUE THREAD

**Issue by OldSneerJaw** [Kashoo/synctos#35]
Add a commandline option to the `make-sync-function` script that, instead of outputting the result as a multi-line JavaScript block, escapes all line ending, backslash and double quote characters so that the output is encased in a single-line JSON-compatible string written to the target file. This option is useful to produce Sync Gateway JSON configurations that don't need to rely on the non-standard backtick/backquote ( ` ) character for multi-line strings in a JSON file ([more info](https://developer.couchbase.com/documentation/mobile/current/guides/sync-gateway/config-properties/index.html#configuration-files)).

For example, invocation should look like the following:

```
./make-sync-function --json-string /path/to/my-doc-definitions.js /path/to/my-new-sync-function.txt
```

or

```
./make-sync-function -j /path/to/my-doc-definitions.js /path/to/my-new-sync-function.txt
```

Tangentially related to #70.

**Comment by OldSneerJaw** [comment]
Reopening to address a bug in which the presence of a backtick in the output causes an error in Sync Gateway when reading the configuration.

**Comment by dkichler** [comment]
Addressed by #260
